api/provider/aws/service_list.go

```diff
@@ -1,110 +1,81 @@
 package aws
 
 import (
-	"strings"
-
 	"github.com/aws/aws-sdk-go/aws"
 	"github.com/aws/aws-sdk-go/service/ecs"
+	"github.com/quintilesims/layer0/common/errors"
 	"github.com/quintilesims/layer0/common/models"
 )
 
 func (s *ServiceProvider) List() ([]models.ServiceSummary, error) {
-	clusterArns, err := s.listClusters()
+	serviceARNs, err := s.listServiceARNs()
 	if err != nil {
 		return nil, err
 	}
 
-	clusterServices := map[*string][]*string{}
-
-	for _, clusterArn := range clusterArns {
-		clusterName := strings.Split(aws.StringValue(clusterArn), ":")[5]
-		clusterName = strings.Replace(clusterName, "cluster/", "", -1)
-
-		if !strings.HasPrefix(clusterName, s.Config.Instance()) {
-			continue
-		}
-
-		serviceArns, err := s.listServices(clusterArn)
+	serviceIDs := make([]string, len(serviceARNs))
+	for i, serviceARN := range serviceARNs {
+		serviceID, err := lookupServiceIDFromServiceARN(s.TagStore, serviceARN)
 		if err != nil {
 			return nil, err
 		}
 
-		clusterServices[clusterArn] = serviceArns
-	}
-
-	serviceSummaries := []models.ServiceSummary{}
-	for _, serviceArns := range clusterServices {
-		for _, serviceArn := range serviceArns {
-			ecsServiceID := s.serviceARNToECSServiceID(aws.StringValue(serviceArn))
-			service := &models.Service{
-				ServiceID: ecsServiceID,
-			}
-			s.populateSummariesTags(service, ecsServiceID)
-
-			summary := models.ServiceSummary{
-				ServiceID:       service.ServiceID,
-				ServiceName:     service.ServiceName,
-				EnvironmentID:   service.EnvironmentID,
-				EnvironmentName: service.EnvironmentName,
-			}
-
-			serviceSummaries = append(serviceSummaries, summary)
-		}
+		serviceIDs[i] = serviceID
 	}
 
-	return serviceSummaries, nil
+	return s.makeServiceSummaryModels(serviceIDs)
 }
 
-func (s *ServiceProvider) listServices(clusterArn *string) ([]*string, error) {
-	serviceArns := []*string{}
-	input := &ecs.ListServicesInput{
-		Cluster: clusterArn,
-	}
-
-	for {
-		output, err := s.AWS.ECS.ListServices(input)
-		if err != nil {
-			return nil, err
+func (s *ServiceProvider) listServiceARNs() ([]string, error) {
+	var serviceARNs []string
+	fn := func(output *ecs.ListServicesOutput, lastPage bool) bool {
+		for _, serviceARN := range output.ServiceArns {
+			serviceARNs = append(serviceARNs, aws.StringValue(serviceARN))
```

Comment:
You need to filter these by layer0 instance

Reply:
✅